### ISSUE TITLE
Support for trino functions like FILTER, ELEMENT_AT, ANY_MATCH

### ISSUE THREAD

**Issue by mauro-lanza** [sqlfluff/sqlfluff#5705]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Currently the following functions i have tried are unparsable:

- FILTER
- ELEMENT_AT
- ANY_MATCH

and others that alter arrays or accept function definitions inside of them.

### Expected Behaviour

it is correctly parsed

### Observed Behaviour

its not parsed

### How to reproduce

```sql
select
    ELEMENT_AT(candidates, 1).id                                                                    AS id,
    COALESCE(object_id, ELEMENT_AT(FILTER(candidates, (x) -> x.id = id), 1).objectid) AS object_id,
    ANY_MATCH(candidates, (x) -> x.id = id)            AS is_winner
from table
```

### Dialect

trino

### Version

3.0.2

### Configuration

[sqlfluff]
dialect = trino
templater = jinja
max_line_length = 0
exclude_rules = ST06
fix_even_unparsable = true

[sqlfluff:indentation]
allow_implicit_indents = True
indented_on_contents = False
indented_ctes = True

[sqlfluff:layout:type:alias_expression]
# We want non-default spacing _before_ the alias expressions.
spacing_before = align
# We want to align them within the next outer select clause.
# This means for example that alias expressions within the FROM
# or JOIN clause would _not_ be aligned with them.
align_within = select_clause
# The point at which to stop searching outward for siblings, which
# in this example would likely be the boundary of a CTE. Stopping
# when we hit brackets is usually a good rule of thumb for this
# configuration.
align_scope = bracketed

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper
[sqlfluff:rules:capitalisation.identifiers]
capitalisation_policy = upper
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
